File: day07/day07-input-output-01-cs/Example1.cs
namespace day07_input_output_01_cs;

using Aws = Pulumi.Aws;

public class Example1
{
    public static void Example1_()
    {
        // 建立 Zone
        var zone = new Aws.Route53.Zone("example", new Aws.Route53.ZoneArgs
        {
            Name = "example.com."
        });

        var www = new Aws.Route53.Record("example", new Aws.Route53.RecordArgs
        {
            // 將 zone 的 output 屬性當作 record 的 input 屬性
            ZoneId = zone.ZoneId,
            Name = "www.example.com.",
            Type = "TXT",
            Ttl = 300,
            Records = new[]
            {
                "Hello, World!"
            }
        });
    }
}

File: day07/day07-input-output-01-ts/example1.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

// 建立 Zone
const zone = new aws.route53.Zone("example", {
    name: "example.com.",
});

const www = new aws.route53.Record("example", {
    // 將 zone 的 output 屬性當作 record 的 input 屬性
    zoneId: zone.zoneId,
    name: "www.example.com.",
    type: "TXT",
    records: [
        "Hello, World"
    ]
});
